### PR TITLE
Set GeometryLayout correctly when reading GPX

### LINKED ISSUE #6148: Set GeometryLayout correctly when reading GPX
The reader of `ol.format.GPX` currently always creates geometry with layout `ol.geom.GeometryLayout.XYZM`.

I am not sure if Z/M are ever used in the GPX parser, but I think when reading a GPX file that only has lat/lon trackpoints, the layout of the imported geometry should be `ol.geom.GeometryLayout.XY`.

### PR DESCRIPTION
**Pull request by romanzoller**
Fixes #6148

### CODE REVIEW

File: src/ol/format/gpx.js [hunk 1]
```diff
@@ -80,12 +81,14 @@ ol.format.GPX.appendCoordinate_ = function(flatCoordinates, node, values) {
   if ('ele' in values) {
     flatCoordinates.push(/** @type {number} */ (values['ele']));
     delete values['ele'];
+    layoutOptions['hasZ'] = true;
```
Comment (romanzoller) [inline src/ol/format/gpx.js]: Good point, fixed. Thanks for the quick review.

**Comment by bartvde** [comment]
2 more questions / remarks:

* I guess you looked into generating the correct flatCoordinates instead of modifying them afterwards?
* a test case for XYZM would be good

TIA

File: src/ol/format/gpx.js [hunk 2]
```diff
@@ -94,6 +97,51 @@ ol.format.GPX.appendCoordinate_ = function(flatCoordinates, node, values) {
...
+  } else if (layoutOptions.hasZ) {
+    layout = ol.geom.GeometryLayout.XYZ;
+    stride = 3;
+  } else if (layoutOptions.hasM) {
+    layout = ol.geom.GeometryLayout.XYM;
+    stride = 3;
+  }
+  if (stride !== 4) {
+    var i;
+    for (i = 0; i < flatCoordinates.length / 4; i++) {
```
Comment (tchandelle) [inline src/ol/format/gpx.js]: Or `for (i = 0, ii = flatCoordinates.length; i < ii; i+= stride)` and removing `* stride` below ? Isn't it better, @bartvde ?

**Comment by romanzoller** [comment]
- Yes, basically the problem is that in the beginning, the final geometry layout / stride is not known, so another approach would be to traverse the xml tree twice, once to determine the layout, and once to extract coordinates. But I think the approach I chose is better in terms of code complexity, and comparable in terms of performance (although of course the only way to know that for sure is to implement both and measure).
- Yes, good point, I didn't realize that there was no XYZM case left after my modifications.

File: src/ol/format/gpx.js (hunk 2 above)
Comment (romanzoller) [inline src/ol/format/gpx.js]: fixed

**Comment by bartvde** [comment]
Ah cool, I had missed those test cases since I only looked at the changeset.

This all looks good to me, any objections @openlayers/core to merging this?

**Comment by bartvde** [comment]
thank you @romanzoller

**Comment by romanzoller** [comment]
Thanks for the review!

**Pull request merged**